tm2/pkg/sdk/vm/keeper.go

```diff
@@ -183,91 +183,38 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 }
 
 // Calls calls a public Gno function (for delivertx).
-func (vm *VMKeeper) Call(ctx sdk.Context, msg MsgCall) (res string, err error) {
-	pkgPath := msg.PkgPath // to import
-	fnc := msg.Func
-	store := vm.getGnoStore(ctx)
-	// Get the package and function type.
-	pv := store.GetPackage(pkgPath, false)
-	pl := gno.PackageNodeLocation(pkgPath)
-	pn := store.GetBlockNode(pl).(*gno.PackageNode)
-	ft := pn.GetStaticTypeOf(store, gno.Name(fnc)).(*gno.FuncType)
-	// Make main Package with imports.
-	mpn := gno.NewPackageNode("main", "main", nil)
-	mpn.Define("pkg", gno.TypedValue{T: &gno.PackageType{}, V: pv})
-	mpv := mpn.NewPackage()
-	// Parse expression.
-	argslist := ""
-	for i := range msg.Args {
-		if i > 0 {
-			argslist += ","
-		}
-		argslist += fmt.Sprintf("arg%d", i)
-	}
-	expr := fmt.Sprintf(`pkg.%s(%s)`, fnc, argslist)
-	xn := gno.MustParseExpr(expr)
+func (vm *VMKeeper) Call(ctx sdk.Context, msg MsgCall) sdk.Result {
+	w := NewWrapper(msg, vm.getGnoStore(ctx))
```

Comment:
Needs a better name.
"Wrapper" is used for things like "CacheWrapper", which actually wraps a singular thing and keeps the method signature the same. In that sense this isn't a "wrapper".